### ISSUE TITLE
"valid-type" & "Invalid base class" with decorators for classes

### ISSUE THREAD

**Issue by dpinol** [python/mypy#17578]
**Bug Report**
mypy reports many false negative errors when defining a class decorator for classes.

**To Reproduce**

```python
import functools
from typing import TypeVar, Any

T = TypeVar("T")


def my_class_decorator(cls: type[T]) -> type[T]:
    @functools.wraps(cls, updated=())
    class Wrapper(cls):
        def __init__(self, *args: Any, **kwargs: Any):
            self.wrapped = cls(*args, **kwargs)

    return Wrapper


@my_class_decorator
class MyClass:
    pass

MyClass()

```

**Expected Behavior**

Mypy should not provide any error

**Actual Behavior**

```python
kk.py:9: error: Variable "cls" is not valid as a type  [valid-type]
        class Wrapper(cls):
                      ^
kk.py:9: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
kk.py:9: error: Invalid base class "cls"  [misc]
        class Wrapper(cls):
                      ^
Found 2 errors in 1 file (checked 1 source file)

```

**Your Environment**

- Mypy version used: mypy 1.11.0 (compiled: yes)
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.12.3

**Comment by sobolevn** [comment]
This is not a false negative. MyPy needs `cls` to be statically defined, otherwise, it won't be able to construct MRO for it.

Use `# type: ignore` for this case.
